### ISSUE TITLE
A function to obtain different licenses for each Github tag

### ISSUE THREAD

**Issue by yongyongsi** [licensee/licensee#664]
### Is your feature request related to a problem? Please describe the problem you're trying to solve.

It appears that in some cases, the license may vary for each tag or commit hash.
this Github repository example provided below, this situation is evident.

https://github.com/platonai/PulsarRPA/tree/0.5.0
https://github.com/platonai/PulsarRPA

 I'm currently using Licensee in a Docker environment through a CI command, but even when I provide different arguments for each commit hash, the license for the 'master' branch is consistently displayed.

### Describe the solution you'd like

I wish it were possible to obtain the license associated with a specific GitHub tag when provided with that tag.

### Describe alternatives you've considered

It would be great if we could retrieve the license using a link that includes a tag or commit hash or by providing the tag as a separate argument.

**Comment by mlinksva** [comment]
> I'm currently using Licensee in a Docker environment through a CI command, but even when I provide different arguments for each commit hash, the license for the 'master' branch is consistently displayed.

How are you providing those arguments? For a remote repo? Or one that you've locally cloned? Can you give an example of the invocation?

**Comment by yongyongsi** [comment]
1. Clone this project to my local pc
`git clone https://github.com/licensee/licensee && cd licensee`

2. docker build
`docker build . --tag licensee

3. execute command
`docker run licensee detect {owner/repo} --remote

In this case, when I included a link with the version to obstain a specific version of the license, it provided the license for the 'master' version.

`docker run licensee detect https://github.com/platonai/PulsarRPA/blob/v1.7.0/ --remote`
When i executed the command for version 1.7.0 of this project. it displayed the AGPL license from the 'master' brance, even though vesion 1.7.0 is supposed to have the Apache License.

`docker run licensee detect https://github.com/platonai/PulsarRPA/blob/0.5.0/ --remote`
Even for version 0.5.0, the license information is present in the 'COPYING' file, but it is incorrectly marked as the AGPL License in the 'LICENSE' file.

So, I cloned both of these projects locally and then ran 'licensee', confirming that the licenses for each project are displayed correctly.
What I want is for the licenses to be correctly displayed with just the version information provided in the link, without the need to clone the projects separately.

**Comment by mlinksva** [comment]
Thanks for explaining. Currently licensee extracts `owner/repo` and ignores anything trailing:

https://github.com/licensee/licensee/blob/11d9b83164ca68d4b75298970a0c1ab6780950c5/lib/licensee/projects/github_project.rb#L17-L23

It should probably either give an error if there is trailing material, or (per your request, but more work) parse it and presumably make Octokit invocations that use this info to switch to the desired branch (I haven't looked into whether this is possible).

Welcome a spike in either direction.

**Comment by mlinksva** [comment]
Fixed by #722:

```
$ bundle exec licensee detect https://github.com/platonai/PulsarRPA
License:        AGPL-3.0
Matched files:  LICENSE
LICENSE:
  Content hash:  10486392ac9e730ec8f77bd1233edfc2d47dd23c
  Confidence:    100.00%
  Matcher:       Licensee::Matchers::Exact
  License:       AGPL-3.0
$ bundle exec licensee detect --ref=v1.7.0 https://github.com/platonai/PulsarRPA
License:        Apache-2.0
Matched files:  LICENSE
LICENSE:
  Content hash:  bec905d850e7f5dc2e2db78a950d4a9db560a0b8
  Confidence:    100.00%
  Matcher:       Licensee::Matchers::Exact
  License:       Apache-2.0
```
